test/test.js

```diff
@@ -1319,6 +1319,20 @@ describe('node-fetch', () => {
 		});
 	});
 
+	it('should create http.Agent class if options.agent is provided as an object', function() {
```

Comment:
@ahmadnassri, agreed. I'll try to merge this PR later today. Thanks for your work!